.github/workflows/publish.yml

```diff
@@ -1,7 +1,10 @@
 name: Build & publish module to registry
+
 on:
-  release:
-    types: [published]
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
```

Comment:
Reminder: set this back the way it used to be before merging. I changed it so the actions would run every time I pushed a commit, which helped debugging.